client/fingerprint/network_unix.go

```diff
@@ -20,54 +19,64 @@ import (
 
 // NetworkFingerprint is used to fingerprint the Network capabilities of a node
 type NetworkFingerprint struct {
-	logger *log.Logger
+	logger            *log.Logger
+	interfaceDetector NetworkInterfaceDetector
+}
+
+// An interface to isolate calls to various api in net package
+// This facilitates testing where we can implement
+// fake interfaces and addresses to test varios code paths
+type NetworkInterfaceDetector interface {
+	Interfaces() ([]net.Interface, error)
+	InterfaceByName(name string) (*net.Interface, error)
+	Addrs(intf *net.Interface) ([]net.Addr, error)
+}
+
+// Implements the interface detector which calls net directly
+type BasicNetworkInterfaceDetector struct {
+}
+
+func (b *BasicNetworkInterfaceDetector) Interfaces() ([]net.Interface, error) {
```

Comment:
Lets call it DefaultNetworkInterfaceDetector.

Basic makes it seem like it doesn't do much!